### ISSUE TITLE
[Python] [LGTM.com] False-positive: Unreachable statement: #2206 sequel

### ISSUE THREAD

**Issue by webknjaz** [Semmle/ql#2351]
**Description of the false positive**

Same as #2206. Except the reproducer contains several subsequent `suppress()`es:
```python
import contextlib

def some_func():
    with contextlib.suppress(KeyError):
        return dict()['a_key']

    with contextlib.suppress(KeyError):  # LGTM complains that it's unreachable
        return dict()['b_key']

    with contextlib.suppress(KeyError):
        return dict()['c_key']
    return 'fallback'
```

So I guess #2078 needs more test cases maybe...

**URL to the alert on the project page on LGTM.com**

https://lgtm.com/projects/g/cherrypy/cheroot/snapshot/0c678bab518ca17151540b7d744faca49b941763/files/cheroot/ssl/builtin.py#x46e331ef60ba4bf1:1

**Comment by tausbn** [comment]
This was an interesting case. The problem here is that we are not identifying the calls to `suppress`  as uses of `contextlib.suppress`. If we follow the definition of `suppress`, we see that it comes from a compatibility library that -- in the absence of `contextlib.suppress` (e.g. if the version of Python is too old) -- defines it itself as a fallback.

As it turns out, we are analysing this project as a Python 2.7 project, so this explains why the fallback version of `suppress` is used. 

In this case, I think there's a more general solution that can be applied, without resorting to looking for specific well-known context handlers (which can fail, as the present example shows). Hopefully we should be able to roll out a fix soon.

Thanks for the report!

**Comment by webknjaz** [comment]
> In this case, I think there's a more general solution that can be applied, without resorting to looking for specific well-known context handlers (which can fail, as the present example shows).

Right, any context manager can choose to suppress exceptions via its `__exit__()` method.

**Comment by webknjaz** [comment]
Got another instance of this: https://lgtm.com/projects/g/ansible/ansible-lint/snapshot/4741210b9feefe9ed54c59a731c1636fd648400e/files/lib/ansiblelint/utils.py?sort=name&dir=ASC&mode=heatmap#x92b14f87ccfa33b7:1

Basically, the same except it uses `continue` in place of `return`:
```python
for ...:
    contextlib.suppress(Exc):
        ...
        continue

    some_code  #  <-- marked as unreachable but in fact, is totally reachable
```
